v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriterDoFn.java

```diff
@@ -184,7 +185,11 @@ public void processElement(ProcessContext c) {
                     }
 
                     // Apply shadow and data table mutations.
-                    transaction.buffer(changeEventContext.getMutations());
+                    if (isBackfill) {
+                      transaction.buffer(changeEventContext.getDataTableMutations());
+                    } else {
+                      transaction.buffer(changeEventContext.getMutations());
+                    }
```

Comment:
Add unit test. 

Also check if the ITs need to be updated or a new one needs to be added.